### PR TITLE
refactor: Abstract deployment logic

### PR DESCRIPTION
**Pull request by marekfexa-outreach**
## What this PR does / why we need it
- Extract logic related to deployment to Custom (formally legacy) or Bootstrap to separate interface to improve code structure

## Jira ID

[XX-XX]

### CODE REVIEW

File: internal/app/config.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
+// Copyright 2022 Outreach Corporation. All Rights Reserved.
+
+// Description: This file contains reading and parsing of devenv config
+
+package app
```
Comment (marekfexa-outreach) [inline internal/app/config.go:5]: This is code extraction from app.go file, no change inside

File: internal/app/app.go [hunk 2]
```diff
@@ -63,6 +62,9 @@ type App struct {
 	// Type is the type of application this is
 	Type Type
 
+	// Abstraction over Custom or Bootstrap deployment
```
Comment (jaredallard) [inline internal/app/app.go:65]: nit: Should start with the name of the field

```suggestion
	// Deployment is an abstraction over Custom or Bootstrap deployment
```

File: internal/app/deployment.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Copyright 2022 Outreach Corporation. All Rights Reserved.
+
+// Description: This file app functions related to app deployment
+
+package app
+
+import "context"
+
```
Comment (jaredallard) [inline internal/app/deployment.go:8]: nit: Missing comment on interface

File: internal/app/bootstrapDeployment.go
Comment (jaredallard) [inline internal/app/bootstrapDeployment.go:1]: I would suggest making structure like so:

```
deployments/<type>/<type.go>
```

Then use `<type>` (.e.g., `custom`) as the name of the package.

Comment (jaredallard) [reply internal/app/bootstrapDeployment.go:1]: You can do it this way as well, I just like the separation of package space (easier to test and easier to read IMO)

**Comment by jaredallard** [comment]
@marekfexa-outreach Can this be closed?

**Comment by jaredallard** [comment]
Re-open if needed!

**Pull request closed without merging**